src/PowerShellEditorServices/Hosting/EditorServicesHost.cs

```diff
@@ -378,7 +389,12 @@ public void StopServices()
         public void WaitForCompletion()
         {
             // TODO: We need a way to know when to complete this task!
-            _languageServer.WaitForShutdown().Wait();
+            if (_languageServer != null)
+            {
+                _languageServer.WaitForShutdown().Wait();
+                return;
```

Comment:
If there's both a languageServer and a debugServer, why do we only wait for one? Possible to document? Just from reading this code block there's no obvious link to the temp session stuff, but I'm guessing there is and we should make sure that's clear

Reply:
Yeah I'll throw a comment. It _is_ related to temp debugging because in temp debugging there is not language server... so we need to wait on something - aka the debug. And in the other scenario, if the language server stops, we wanna stop the whole process

Reply:
added.